net/sched/sch_dsprio.c

```diff
@@ -138,13 +140,16 @@ static int dsprio_enqueue(struct sk_buff *skb, struct Qdisc *sch,
 			q->lowest_prio = prio;
 
 		sch->q.qlen++;
+		q->qstats[prio].qlen++;
```

Comment:
We don't need to update `q->qstats[prio].qlen` because this information is already available in array `qdiscs` of `struct dsprio_sched_data`.

Make adjusts throughout the code to account for this change.